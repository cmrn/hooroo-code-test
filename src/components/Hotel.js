import React from 'react';
import PropTypes from 'prop-types';
import cnames from 'classnames';
import './Hotel.css';
import Rating from './Rating';

const Hotel = ({ hotel, className, ...props }) => {
  return (
    <div className={cnames("Hotel", className)} {...props}>
      <div className="Hotel-image">
        <span className="Hotel-promotion">{hotel.promotion}</span>
        <img src={hotel.image} alt='' />
      </div>
      <div className="Hotel-details">
        <div className="Hotel-title">
          <h3>{hotel.title}</h3>
          <Rating className="Hotel-rating" rating={hotel.rating} type={hotel.ratingType} />
        </div>
        <div className="Hotel-address">{hotel.address}</div>
        <div className="Hotel-room">{hotel.roomName}</div>
      </div>
      { hotel.freeCancellation ? <div className="Hotel-cancellation">Free cancellation</div> : null }
      <div className="Hotel-price">
        <div className="Hotel-priceDescription">1 night total (AUD)</div>
        <div>
          <span className="Hotel-currencySymbol">$</span> <span className="Hotel-priceNumber">{hotel.price}</span>
        </div>
        { hotel.savings > 0 ? <div className="Hotel-savings">Save ${hotel.savings}~</div> : null }
      </div>
    </div>
  )
};

Hotel.propTypes = {
  hotel: PropTypes.object.isRequired, // TODO: Make a shape for hotel
  className: PropTypes.string,
};

export default Hotel;