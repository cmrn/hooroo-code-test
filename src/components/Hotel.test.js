import React from 'react';
import renderer from 'react-test-renderer';
import Hotel from './Hotel';

beforeEach(() => {
  Math.random = jest.fn(() => 0.0487723387081509); // react-star uses a random number
})

const hotel = {
  "id": "cxd650nuyo",
  "title": "Courtyard by Marriott Sydney-North Ryde",
  "address": "7-11 Talavera Rd, North Ryde",
  "image": "https://unsplash.it/145/125/?random1",
  "rating": 4,
  "ratingType": "self",
  "promotion": "Exclusive Deal",
  "roomName": "Deluxe Balcony Room",
  "price": 329,
  "savings": 30,
  "freeCancellation": true
};

it('renders a hotel', () => {
  const tree = renderer.create(
    <Hotel hotel={hotel} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('allows a className to be set', () => {
  const tree = renderer.create(
    <Hotel hotel={hotel} className="SomeClass" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('passes through other props', () => {
  const tree = renderer.create(
    <Hotel hotel={hotel} aria-label="another prop" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});