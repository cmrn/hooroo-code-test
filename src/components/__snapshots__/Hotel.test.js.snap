// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows a className to be set 1`] = `
<div
  className="Hotel SomeClass"
>
  <div
    className="Hotel-image"
  >
    <span
      className="Hotel-promotion"
    >
      Exclusive Deal
    </span>
    <img
      alt=""
      src="https://unsplash.it/145/125/?random1"
    />
  </div>
  <div
    className="Hotel-details"
  >
    <div
      className="Hotel-title"
    >
      <h3>
        Courtyard by Marriott Sydney-North Ryde
      </h3>
      <div
        aria-label="4 stars (self rated)"
        className="Hotel-rating Rating"
      >
        <div
          className={undefined}
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "
                  .react-stars-00487723387081509:before {
                    position: absolute;
                    overflow: hidden;
                    display: block;
                    z-index: 1;
                    top: 0; left: 0;
                    width: 50%;
                    content: attr(data-forhalf);
                    color: #ffd700;
                }",
              }
            }
          />
          <span
            className=""
            data-forhalf="●"
            data-index={0}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={1}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={2}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={3}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={4}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ccc",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
        </div>
      </div>
    </div>
    <div
      className="Hotel-address"
    >
      7-11 Talavera Rd, North Ryde
    </div>
    <div
      className="Hotel-room"
    >
      Deluxe Balcony Room
    </div>
  </div>
  <div
    className="Hotel-cancellation"
  >
    Free cancellation
  </div>
  <div
    className="Hotel-price"
  >
    <div
      className="Hotel-priceDescription"
    >
      1 night total (AUD)
    </div>
    <div>
      <span
        className="Hotel-currencySymbol"
      >
        $
      </span>
       
      <span
        className="Hotel-priceNumber"
      >
        329
      </span>
    </div>
    <div
      className="Hotel-savings"
    >
      Save $
      30
      ~
    </div>
  </div>
</div>
`;

exports[`passes through other props 1`] = `
<div
  aria-label="another prop"
  className="Hotel"
>
  <div
    className="Hotel-image"
  >
    <span
      className="Hotel-promotion"
    >
      Exclusive Deal
    </span>
    <img
      alt=""
      src="https://unsplash.it/145/125/?random1"
    />
  </div>
  <div
    className="Hotel-details"
  >
    <div
      className="Hotel-title"
    >
      <h3>
        Courtyard by Marriott Sydney-North Ryde
      </h3>
      <div
        aria-label="4 stars (self rated)"
        className="Hotel-rating Rating"
      >
        <div
          className={undefined}
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "
                  .react-stars-00487723387081509:before {
                    position: absolute;
                    overflow: hidden;
                    display: block;
                    z-index: 1;
                    top: 0; left: 0;
                    width: 50%;
                    content: attr(data-forhalf);
                    color: #ffd700;
                }",
              }
            }
          />
          <span
            className=""
            data-forhalf="●"
            data-index={0}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={1}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={2}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={3}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={4}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ccc",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
        </div>
      </div>
    </div>
    <div
      className="Hotel-address"
    >
      7-11 Talavera Rd, North Ryde
    </div>
    <div
      className="Hotel-room"
    >
      Deluxe Balcony Room
    </div>
  </div>
  <div
    className="Hotel-cancellation"
  >
    Free cancellation
  </div>
  <div
    className="Hotel-price"
  >
    <div
      className="Hotel-priceDescription"
    >
      1 night total (AUD)
    </div>
    <div>
      <span
        className="Hotel-currencySymbol"
      >
        $
      </span>
       
      <span
        className="Hotel-priceNumber"
      >
        329
      </span>
    </div>
    <div
      className="Hotel-savings"
    >
      Save $
      30
      ~
    </div>
  </div>
</div>
`;

exports[`renders a hotel 1`] = `
<div
  className="Hotel"
>
  <div
    className="Hotel-image"
  >
    <span
      className="Hotel-promotion"
    >
      Exclusive Deal
    </span>
    <img
      alt=""
      src="https://unsplash.it/145/125/?random1"
    />
  </div>
  <div
    className="Hotel-details"
  >
    <div
      className="Hotel-title"
    >
      <h3>
        Courtyard by Marriott Sydney-North Ryde
      </h3>
      <div
        aria-label="4 stars (self rated)"
        className="Hotel-rating Rating"
      >
        <div
          className={undefined}
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "
                  .react-stars-00487723387081509:before {
                    position: absolute;
                    overflow: hidden;
                    display: block;
                    z-index: 1;
                    top: 0; left: 0;
                    width: 50%;
                    content: attr(data-forhalf);
                    color: #ffd700;
                }",
              }
            }
          />
          <span
            className=""
            data-forhalf="●"
            data-index={0}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={1}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={2}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={3}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ffd700",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
          <span
            className=""
            data-forhalf="●"
            data-index={4}
            onClick={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "color": "#ccc",
                "cursor": "default",
                "display": "block",
                "float": "left",
                "fontSize": "30px",
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            ●
          </span>
        </div>
      </div>
    </div>
    <div
      className="Hotel-address"
    >
      7-11 Talavera Rd, North Ryde
    </div>
    <div
      className="Hotel-room"
    >
      Deluxe Balcony Room
    </div>
  </div>
  <div
    className="Hotel-cancellation"
  >
    Free cancellation
  </div>
  <div
    className="Hotel-price"
  >
    <div
      className="Hotel-priceDescription"
    >
      1 night total (AUD)
    </div>
    <div>
      <span
        className="Hotel-currencySymbol"
      >
        $
      </span>
       
      <span
        className="Hotel-priceNumber"
      >
        329
      </span>
    </div>
    <div
      className="Hotel-savings"
    >
      Save $
      30
      ~
    </div>
  </div>
</div>
`;
